### ISSUE TITLE
Can we align GLM-130B to human like chatgpt? 

### ISSUE THREAD

**Issue by AnShengqiang** [THUDM/GLM-130B#43]

**Comment by Xiao9905** [comment]
Certainly. The alignment for GLM-130B could be important, and we are on preliminary surveying.

**Comment by conceptofmind** [comment]
You could use the current glm-10b on huggingface with trl/trlx to construct a model with rlhf.

**Comment by smeyerhot** [comment]
What is trl/trlx? I am very interested in this use case. Why must the 10-b parameter model be used for rlhf?

**Comment by smeyerhot** [comment]
I am actively working on this task and would be very interested in further development coordination.

**Comment by conceptofmind** [comment]
@smeyerhot Trl is Transformer Reinforcement Learning a library built by Huggingface for training language models with PPO. Trlx is an extension of Trl built by CarperAI. Both cover the same use-case for training models using reinforcement learning with human feedback. You can also build the same functionality with actor-critic ppo in PyTorch although it would require more extensive domain knowledge. You do not have to use glm-10b but it is publicly available on Huggingface's model hub unlike 130b which requires you to apply for access. You can use any encoder-decoder or decoder-only model. We are on an issue relating to GLM for aligning human feedback with the model which is why I suggested using the 10b parameter one.

**Comment by Syno8** [comment]
chatgpt can generate the format text and image. this need to keep the pertaining data in original format

**Comment by beautifull4frank** [comment]
hi gays, I use bloom to implement ppo successfully. 
![image](https://user-images.githubusercontent.com/127466507/224220085-28d24f1f-b34e-4851-80b7-17d16f3625fe.png)
But I found the Bloom model use the AutoModelForCausalLM function.
![image](https://user-images.githubusercontent.com/127466507/224220320-fb6ef425-9d78-4d4c-b156-608ed6991ff0.png)

however, the glm is using the AutoModelForSeq2SeqLM function.
![image](https://user-images.githubusercontent.com/127466507/224220799-1e906d2f-4ef6-4caf-a1c0-98092d9a316b.png)
there is no LM in AutoModelForSeq2SeqLM  model, so do u know how to correct ?
